### PR TITLE
feat(payment): PAYPAL-1888 added PayPalCommerceButtonStrategy to paypal-commerce-integration package

### PR DESCRIPTION
**Pull request by serhii-tkachenko**
## What?
Added PayPalCommerceButtonStrategy to paypal-commerce-integration package

## Why?
As part of PayPalCommerce code migration process

## Testing / Proof
Unit tests
Manual tests

### CODE REVIEW

**Review by PavlenkoM — APPROVED** [review]

File: packages/paypal-commerce-integration/src/utils/get-valid-button-style.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+function getValidTagline(tagline?: boolean, layout?: string): boolean | undefined {
+    if (tagline && typeof tagline === 'boolean' && layout === StyleButtonLayout.horizontal) {
+        return tagline;
+    }
+
+    return undefined;
+}
+
+function getValidHeight(height?: number): number {
+    const defaultHeight = 40;
```
Comment (vitalii-koshovyi) [inline packages/paypal-commerce-integration/src/utils/get-valid-button-style.ts:54]: we can use it as default argument value:

`function getValidHeight(height?: number = 40): number {`

Comment (vitalii-koshovyi) [reply packages/paypal-commerce-integration/src/utils/get-valid-button-style.ts:54]: it's not required, just recommendation

**Review by vitalii-koshovyi — APPROVED** [review]

File: packages/paypal-commerce-integration/src/utils/get-valid-button-style.ts (hunk 1 above)
Comment (vitalii-koshovyi) [reply packages/paypal-commerce-integration/src/utils/get-valid-button-style.ts:54]: oh my bad. everything fine👍

File: packages/paypal-commerce-integration/src/paypal-commerce/paypal-commerce-button-strategy.ts [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+        return this.formPoster.postForm('/checkout.php', {
+            payment_type: 'paypal',
+            action: 'set_external_checkout',
+            provider: methodId,
+            order_id: orderId,
+            ...(cart.source === CartSource.BuyNow && { cart_id: cart.id }),
+        });
+    }
+
+    private getPayPalCommerceSdkOrThrow(): PayPalSDK {
```
Comment (serhii-tkachenko) [inline packages/paypal-commerce-integration/src/paypal-commerce/paypal-commerce-button-strategy.ts:387]: TODO: update method name to `getPayPalSdkOrThrow`

**Pull request merged**